[Teng Chin Chong] iP

iamanoob44 opened the pull request:
# Hi! Welcome to *Shagbot!*
> "Life can be tough, difficult and totally unpredictable but NOT impossible." – ([source](https://medium.com/@satvsonia/life-can-be-tough-difficult-and-totally-unpredictable-but-not-impossible-6fa59a79c8af))

Yes, I hear you. Life is already tiring and shag with the constant workflow – assignments, project deadlines, meeting up with friends, gaming events, commitments and so much more. Just thinking about how to track all of these tasks at once sounds so shag...

## Introducing ***ShagBot***!
Your **personalised chatbot assistant**, designed to help you keep track and manage your tasks efficiently and make your life less shag! 

***Shagbot*** is : 

* text-based
* user-friendly
* easy to learn, use and navigate
* ~~FAST~~ *INCREDIBLY FAST* to use

### How to access Shagbot?
All you need to do is,

1. Download it from the files [here]().
2. Go to Launcher.java and run the program

```Java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```


3. Add your tasks.
4. Let it manage your tasks for you😊

And it is **FREE AND EASY TO USE!**


#### Features of ***Shagbot***
 - [X] Add or Delete your tasks
 - [X] Track your tasks using keywords
 - [X] Find any tasks to complete based on their time periods or deadlines of the tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

iamanoob44 committed:
Rename Duke.java to Shagbot.java

iamanoob44 committed:
Rename Duke.java to Shagbot.java

iamanoob44 committed:
Add greeting and exit

iamanoob44 committed:
Add Ui class for greeting, exit and echo

iamanoob44 committed:
Refine greeting and exit

iamanoob44 committed:
Incorporate echo feature in Shagbot class

iamanoob44 committed:
Add Task class to support add and store

iamanoob44 committed:
Include add and list tasks features

iamanoob44 committed:
Rename original Task class to ManageTasks class

iamanoob44 committed:
Add Parser class for parsing commands

iamanoob44 committed:
Add mark/unmark features

iamanoob44 committed:
Add Todo class to handle todo feature

iamanoob44 committed:
Add Deadline class to handle deadline feature

iamanoob44 committed:
Add Event class to handle event feature

iamanoob44 committed:
Add support for todo, event and deadlines

iamanoob44 committed:
Automated testing of Shagbot

iamanoob44 committed:
Add support for handling exceptions

iamanoob44 committed:
Change Task array to ArrayList

iamanoob44 committed:
Add support for deleting tasks

iamanoob44 committed:
Refine exception handling for deletion

iamanoob44 committed:
Refine exception handling for marking tasks

iamanoob44 committed:
Deadline class: Enhance alignment with coding standards

The Deadline class has formatting inconsistencies and lacks clarity
in documentation. These issues reduce readability and make the code
less maintainable.

This commit improves the class by:
- Updating Javadoc comments for the constructor and toString method
  to make them more descriptive and aligned with coding standards.
- Ensuring consistent formatting, indentation, and layout to enhance
  readability.
- Refining method descriptions to comply with Java documentation
  standards.
- Adding a detailed explanation of the string representation format
  in the toString method.

These changes improve the readability and adherence to coding
standards of the Deadline class.

iamanoob44 committed:
Event class: Enhance alignment with coding standards

The Event class was updated to improve readability and adherence to
coding standards. Changes include:

- Improving Javadoc comments for the constructor and toString method.
- Ensuring consistent formatting, indentation, and layout.
- Refining method descriptions to comply with Java documentation
  standards.
- Adding a detailed explanation of the string representation format
  in the toString method.

These changes enhance the readability and consistency of the code.

iamanoob44 committed:
ManageTasks: Enhance alignment with coding standards

The ManageTasks class contains formatting inconsistencies and unclear
documentation, which reduce readability.

This commit addresses these issues by:
- Updating Javadoc comments for the constructor and other methods
  to make them more descriptive and align with coding standards.
- Ensuring consistent formatting, indentation, and layout to improve
  readability.
- Refining method descriptions to comply with Java documentation
  standards.

Additionally, this commit adjusts the Deadline and Event classes to
improve consistency and alignment with coding standards.

These updates enhance the readability and adherence to coding
standards of the ManageTasks class.

iamanoob44 committed:
Parser: Enhance alignment with coding standards

The Parser class contains formatting inconsistencies and unclear
documentation, which reduce readability.

This commit addresses these issues by:
- Updating Javadoc comments for the constructor and other methods
  to make them more descriptive and align with coding standards.
- Ensuring consistent formatting, indentation, and layout to improve
  readability.

These updates enhance the readability and adherence to coding
standards of the Parser class.

iamanoob44 committed:
Shagbot: Enhance alignment with coding standards

Improve the Shagbot class by addressing formatting inconsistencies
and enhancing Javadoc comments for better adherence to coding
standards.

Changes include:
- Refining Javadoc comments for the constructor and methods to ensure
  clarity and compliance with Java coding standards.
- Standardise formatting, indentation, and layout for better
  readability.
- Update method descriptions to better align with Java conventional
  documentation.

These updates improve readability and compliance with coding standards.

iamanoob44 committed:
ShagBotException: Enhance alignment with coding standards

Update the constructor Javadoc comments to fully align with Java coding
standards by adding descriptions of its purpose and usage.

Improve readability and ensure compliance with Java documentation best
practices.

iamanoob44 committed:
Task: Enhance alignment with coding standards

Update the Javadoc comments to fully align with Java coding
standards by adding periods at the end of sentences and
capitalising the first letter of the javadoc comments.

These updates improve readability and compliance with coding standards.

iamanoob44 committed:
Todo class: Enhance alignment with coding standards

Improve the Todo class by addressing formatting inconsistencies
and enhancing Javadoc comments for better adherence to coding
standards.

Changes include:
- Refining Javadoc comments for the constructor and the toString method
  to ensure clarity and compliance with Java coding standards.
- Standardise formatting, indentation, and layout for better
  readability.
- Update method descriptions to better align with Java conventional
  documentation.

These updates improve readability and compliance with coding standards.

iamanoob44 committed:
Ui class: Enhance alignment with coding standards

This commit improves the Ui class by addressing formatting
inconsistencies and unclear documentation.

Key updates include:
- Refining Javadoc comments for the constructor and methods
  to make them more descriptive and informative.
- Improve formatting, indentation, and layout to ensure consistency.
- Introduce a static final constant, LINE_SEPARATOR, to eliminate
  repetitive string literals and enhance code maintainability.

These changes make the Ui class easier to read, understand,
and maintain while more closely adhering to the Java coding standards.

iamanoob44 committed:
Add Storage class to handle saving of tasks

iamanoob44 committed:
Add a getter function in Deadline.java

iamanoob44 committed:
Add getter functions in Event.java

iamanoob44 committed:
Add getter functions in Task.java

iamanoob44 committed:
Add support for saving tasks feature

iamanoob44 committed:
Ignore data.txt file to track saved tasks

iamanoob44 committed:
Merge branch 'branch-Level-7'

iamanoob44 committed:
Ignore dataoftasks.txt file for saved tasks

iamanoob44 committed:
Utilise new txt file for saved tasks

iamanoob44 committed:
Add LocalDateTime for date and time handling

This helps support for parsing, formatting, and managing
date and time features in the Shagbot application.

iamanoob44 committed:
Merge branch 'branch-Level-8'

iamanoob44 committed:
Rename ManageTasks.java to TaskList.java

iamanoob44 committed:
Refactor ManageTasks to TaskList for naming and OOP design

iamanoob44 committed:
Refactor ManageTasks to TaskList and fix bug

iamanoob44 committed:
Merge branch 'A-MoreOOP'

iamanoob44 committed:
Add Shagbot package for classes

iamanoob44 committed:
Merge branch 'branch-A-Packages'

iamanoob44 committed:
Merge branch 'add-gradle-support'

iamanoob44 committed:
Add multiple packages to organise classes

iamanoob44 committed:
Move TaskList class from util to tasks package

iamanoob44 committed:
Create JUnit tests for TaskList and Parser classes

iamanoob44 committed:
Merge branch 'branch-A-JUnit'

iamanoob44 committed:
Update build.gradle file

iamanoob44 committed:
Merge branch 'branch-A-Gradle'

iamanoob44 committed:
Remove checkstyle plugin

iamanoob44 committed:
Update build.gradle file again

iamanoob44 committed:
Add more support for Junit Testing

iamanoob44 committed:
Refine test cases for JUnit testing

iamanoob44 committed:
Ignore changes to META-INF while creating jar file

iamanoob44 committed:
Merge branch 'branch-A-Jar'

iamanoob44 committed:
Add and refine JavaDoc comments

iamanoob44 committed:
Align coding standards for classes in tasks package

iamanoob44 committed:
Align coding standards for classes in tasks package again

iamanoob44 committed:
Align coding standards for classes in util package

iamanoob44 committed:
Align coding standards for Shagbot.java

iamanoob44 committed:
Add 'find' command support to Parser.java

iamanoob44 committed:
Add 'find' command support to Ui.java

iamanoob44 committed:
Merge branch 'branch-A-JavaDoc'

iamanoob44 committed:
Merge branch 'branch-A-CodingStandard'

# Resolve Merge Conflicts:
#	src/main/java/shagbot/Shagbot.java
#	src/main/java/shagbot/tasks/Deadline.java
#	src/main/java/shagbot/tasks/Event.java
#	src/main/java/shagbot/util/Storage.java
#	src/main/java/shagbot/util/Ui.java

iamanoob44 committed:
Merge branch 'branch-Level-9'

iamanoob44 committed:
Reorganise order of imports

iamanoob44 committed:
Replace wildcard imports with specified class imports

iamanoob44 committed:
Refine JavaDoc to align with coding standards

iamanoob44 committed:
Refine JavaDoc comments for this class

iamanoob44 committed:
Add checkstyle configs

iamanoob44 committed:
Add 'checkstyle' plugin

iamanoob44 committed:
Merge branch 'branch-A-CheckStyle'

iamanoob44 committed:
Enforce checkstyle standards in util package

iamanoob44 committed:
Enforce checkstyle standards in tasks package

iamanoob44 committed:
Enforce checkstyle standards in this file

iamanoob44 committed:
Add new dependencies and update mainClassattribute

iamanoob44 committed:
Add Main and Launcher classes for GUI application

iamanoob44 committed:
Add GUI support for Shagbot

iamanoob44 committed:
Add GUI support with FXML for Shagbot

iamanoob44 committed:
Organise these classes into guihelp package

iamanoob44 committed:
Add and update UI colour theme

iamanoob44 committed:
Reformat code to align with coding standard

iamanoob44 committed:
Merge branch 'branch-Level-10'

iamanoob44 committed:
Reformat code to align with coding standards

iamanoob44 committed:
Replace if-else statements with switch-case

iamanoob44 committed:
Fix an assertion bug in a JUnit test

malihahaque commented on src/main/java/shagbot/Main.java:
```diff
@@ -2,24 +2,80 @@
 
 import javafx.application.Application;
```
Code is easy to follow and is clear! Follows good coding standards. Keep it up

malihahaque reviewed (commented):
Good job! Use of OOP made it easy for me to read the code. All the coding standards were maintained

UltimateHG commented on src/main/java/shagbot/util/Parser.java:
```diff
@@ -0,0 +1,283 @@
+     */
+    protected void handleMarkCommand(String command, boolean isMark) throws ShagBotException {
+        try {
+            if (command.trim().equalsIgnoreCase("MARK")
+                    || command.trim().equalsIgnoreCase("UNMARK")) {
```
Would it be better to also handle this in the main switch case? It might be better organisation of the code in this way.

UltimateHG commented on src/main/java/shagbot/Shagbot.java:
```diff
@@ -0,0 +1,110 @@
+        this.ui = new Ui(name);
+        this.taskList = new TaskList();
+        this.parser = new Parser(taskList, ui);
+        this.storage = new Storage("./data/dataoftasks.txt");
```
Would it be better to use a separate variable instead of hardcoding the filepath? i.e. private final SAVEPATH

UltimateHG reviewed (commented):
Easy-to-understand class abstractions, looks good 👍

iamanoob44 committed:
Generate new jar file

iamanoob44 committed:
Add assertions to shagbot package

Numerous classes in the shagbot package lack assertions that check for
invalid inputs and improper states. Erroneous entries may result in
bugs in the functionality of the code.

Let's add assertions to:
- Verify that inputs and configurations are non-null.
- Check that numerical parameters are within acceptable ranges.
- Ensure that internal states align with the expected conditions.

This commit improves error detection during the development phase and
boosts the robustness of the code.

iamanoob44 committed:
Merge pull request #1 from iamanoob44/branch-A-Assertions

Add assertions to shagbot package

iamanoob44 committed:
Merge branch 'master' into branch-A-CodeQuality

iamanoob44 committed:
Improve code quality through refactoring

Some parts of the code in the shagbot package contain redundant logic,
inconsistent naming, and unclear structures such as the repetitive
usage of literals. This reduces code readability and robustness.

Let's improve code quality by:
- Simplifying complex logic and removing redundant checks.
- Using 'Extract Method' and variable declarations through the
  refactoring tool for better clarity.
- Extracting repeated logic into helper functions for reuse.
- Declaring literals as constants for better maintainability.

This commit enhances code quality through refactoring, making the code
easier to read, debug, and maintain while preserving Shagbot’s existing
functionality.

iamanoob44 committed:
Merge pull request #2 from iamanoob44/branch-A-CodeQuality

Merging the branch-A-CodeQuality into master branch

iamanoob44 committed:
Merge branch 'master' into branch-A-Streams

iamanoob44 committed:
Refine code quality further

iamanoob44 committed:
Merge pull request #3 from iamanoob44/branch-A-CodeQuality

Refine code quality

iamanoob44 committed:
Merge branch 'master' into branch-A-Streams

iamanoob44 committed:
Incorporate streams into Parser.java

iamanoob44 committed:
Merge pull request #4 from iamanoob44/branch-A-Streams

Incorporate the Streams feature

iamanoob44 committed:
Change access modifiers of some classes

iamanoob44 committed:
Add support for snooze feature

Add setter methods in the Deadline and Event classes to
support the snooze/reschedule feature. This change enables dynamic
updates of the task timings when the snooze command is used, while
improving encapsulation and code maintainability.

Let's introduce:
- setByTiming(LocalDateTime) in Deadline.java
  to manage the new snooze due dates.
- setStart(LocalDateTime) and setEnd(LocalDateTime) in Event.java
  to handle snooze of new start and end times.
- New methods in Parser.java to support the snooze feature

These updates allow Shagbot to reschedule tasks more flexibly while
adhering to OOP Principles.

iamanoob44 committed:
Merge branch 'branch-BCD-Extension'

iamanoob44 committed:
Add reminder feature

iamanoob44 committed:
Merge branch 'branch-BCD-Extension'

iamanoob44 committed:
Remove dead code from Shagbot.java

iamanoob44 committed:
Remove dead code from Ui.java

iamanoob44 committed:
Refine JavaDoc comments

iamanoob44 committed:
Add package and classes to handle various commands

iamanoob44 committed:
Refactor code: integrate with commands package

iamanoob44 committed:
Improve error handling across multiple classes

iamanoob44 committed:
Merge branch 'branch-A-MoreErrorHandling'

iamanoob44 committed:
Add and refine JUnit tests

iamanoob44 committed:
Modify code to further support Junit testing

iamanoob44 committed:
Merge branch 'branch-A-MoreTesting'

iamanoob44 committed:
Refactor code to improve code quality

iamanoob44 committed:
Change images for GUI

iamanoob44 committed:
Add Ui.png to showcase Shagbot

iamanoob44 committed:
Revert "Add Ui.png to showcase Shagbot"

This reverts commit 7dfae0fbc5ed8933bb19ff9b5eff163a0b6c51c3.

iamanoob44 committed:
Ignore file

iamanoob44 committed:
Add Shagbot user guide

iamanoob44 committed:
Merge branch 'branch-A-UserGuide'

iamanoob44 committed:
Add Ui.png for showcase

iamanoob44 committed:
Update README.md

iamanoob44 committed:
Update the README.md

iamanoob44 committed:
Update README.md due to formatting error

iamanoob44 committed:
Refine README.md

iamanoob44 committed:
Include attribution to images

Include attributions for the shagbots.png and users.png files , in Shagbot's GUI Application.

iamanoob44 committed:
Include output section for tasks creation

iamanoob44 committed:
Update build.gradle file

iamanoob44 committed:
Add support for product name in GUI window

iamanoob44 committed:
Ignore file

iamanoob44 committed:
Revert "Ignore file"

This reverts commit a0f7d4d30dc3b36c22a7ef3f3c599b562ed2d0cc.

iamanoob44 committed:
Update Ui.png file to include product name

iamanoob44 committed:
Refine the README.md document

Fix typo errors in this document.

iamanoob44 committed:
Fix bugs when loading from a corrupt txt file

iamanoob44 committed:
Refactor code to enhance coding standard

iamanoob44 committed:
Improve on the code quality

iamanoob44 committed:
Refine coding standard for Boolean variables

iamanoob44 committed:
Fix performance issues that causes slight lag

iamanoob44 committed:
Edit background colour in main.css file

iamanoob44 committed:
Edit Ui.png for better showcase of Shagbot
